playground/backend/internal/code_processing/code_processing.go

```diff
@@ -406,6 +409,31 @@ func cancelCheck(ctx context.Context, pipelineId uuid.UUID, cancelChannel chan b
 	}
 }
 
+// readGraphFile reads graph from the file and keeps it to the cache.
+// If context is done it means that the code processing was finished (successfully/with error/timeout).
+// Write graph to the cache if this in the file.
+// In other case each pauseDuration checks that graph file exists or not and try to save it to the cache.
+func readGraphFile(pipelineLifeCycleCtx, backgroundCtx context.Context, cacheService cache.Cache, graphFilePath string, pipelineId uuid.UUID) {
+	ticker := time.NewTicker(pauseDuration)
+	for {
+		select {
+		// waiting when graph file appears
+		case <-ticker.C:
+			if _, err := os.Stat(graphFilePath); err == nil {
+				ticker.Stop()
+				utils.ReadAndSetToCacheGraph(backgroundCtx, cacheService, pipelineId, graphFilePath)
```

Comment:
Shouldn't we check if `utils.ReadAndSetToCacheGraph` returns an error and if so – don't stop the ticker? Or we may have such errors only if something definitely went wrong and there is no need to check the graph file again?

Reply:
The second option. Since we already check that file exists - then if we get an error it will not be related to the existence of the file, but rather the other error.